tests/gtests/graph/unit/backend/elyzor/test_compile_execute.cpp

```diff
@@ -105,3 +106,89 @@ TEST(ElyzorGraphTest, CompleteInputShapes) {
 }
 
 #endif
+
+// Verify that 'elyzor' backend is successfully registered
+// and available via the backends registry
+TEST(ElyzorSanityTests, ElyzorBackendRegistration) {
+    std::vector<const graph::backend_t *> &backends
+            = graph::backend_registry_t::get_singleton()
+                      .get_registered_backends();
+    auto elyzor_backend = std::find_if(
+            backends.begin(), backends.end(), [](const graph::backend_t *bkd) {
+                return bkd->get_name() == "elyzor_compiler_backend";
+            });
+    ASSERT_NE(elyzor_backend, backends.end());
+    EXPECT_FLOAT_EQ((*elyzor_backend)->get_priority(), 3.0);
+}
+
+// Verify that 'graph_compiler_loader::get_vtable()' returns
+// vtable with valid function pointers
+TEST(ElyzorSanityTests, CompilerLoaderValidVtable) {
```

Comment:
Will the test fail if there is no graph_compiler lib in the env? I think, there  should be an option to disable the test if the lib is not found by cmake.

Reply:
Yes, this test as well as `MulQuantizeGCAPI`, `MulQuantizeOneDNNGraph` will fail, if there's no `libgraph_compiler.so` available in `LD_LIBRARY_PATH` (basically any non-trivial test would fail, as without the library, elyzor backend is actually non-functional)

> I think, there should be an option to disable the test if the lib is not found by cmake.

We load the library in runtime, so I suggest we make an environment variable that we read in runtime and skip certain tests:

```cpp

#define SKIP_WHEN_NO_LIB_TESTS() \
    if (!graph::utils::getenv_int_internal( \
                "ENABLE_ELYZOR_GRAPH_COMPILER_LIB_TESTS", /*default_value=*/1)) { \
        GTEST_SKIP(); \
        return; \
    }

TEST(ElyzorTest, TestCompile) {
    SKIP_WHEN_NO_LIB_TESTS()

    // test that uses gc api
}

TEST(ElyzorTest, TestSomethingTrivial) {
    // test that doesn't use gc API, so no 'SKIP_...()'
}
```

Then if you run compiled gtests with the variable set to 0, the tests will be skipped:
```bash
$ ./tests/test_graph_unit # run all tests, most of them will fail, if there's no libgraph_compiler.so available
$ ENABLE_ELYZOR_GRAPH_COMPILER_LIB_TESTS=0 ./tests/test_graph_unit # only basic tests
```

Reply:
applied this in b9357af11ba44abef9f76b19def8df20ddb8e149

Reply:
There could be a problem when the tests are run by CI - it will fail because there is no the GC library in the env. Usually it's solved by checking for the dependencies in cmake and adding definitions to CXX_FLAGS. Then, you can wrap the code into `#ifdef` checks.

Currently, it does not make sense to add any findpackage logic into the cmake scripts, but you can wrap the tests into the `ifdef` checks and then add `-DCMAKE_CXX_FLAGS=-DHAS_GRAPH_COMPILER` to your dev env.

Reply:
> There could be a problem when the tests are run by CI - it will fail because there is no the GC library in the env.

So if the lib is not available, we can set the env variable anyway and skip certain tests in runtime. I don't like the idea of a compile-time `-DHAS_GRAPH_COMPILER` flag as it generally makes no sense to compile elyzor backend with GC features disabled (it will only affect the test set). We can use the same binary for testing with & without graph compiler lib by simply switching the env variable.

> Currently, it does not make sense to add any findpackage logic into the cmake scripts

Probably not "currently", but rather "at all". CMake in oneDNN shouldn't check for the existence of `libgraph_compiler.so` as oneDNN is supposed to be compiled separately from graph compiler. They're paired together later and only in runtime.

Reply:
If we add an option to skip failing tests, it means, that the tests fail by default. I.e. if someone downloads the sources and builds oneDNN from the sources, the tests will fail. The option should be opposite - enable the tests if the lib is installed.

If you don't like the compile-time option, you may check an env variable - if it's set and the library is not found - fail, if not set - skip the test.

Reply:
> if someone downloads the sources and builds oneDNN from the sources, the tests will fail

You have to explicitly specify `-DONEDNN_EXPERIMENTAL_ELYZOR_BACKEND=ON` to build elyzor backend in oneDNN, so if someone builds oneDNN with default parameters no tests will fail. When one explicitly specifies the flag to build elyzor backend, they should be aware of what they're doing and that they will receive a <i>runtime error</i> if `dlopen` fails to find the library.

As said earlier, the elyzor backend is practically non-functional without `libgraph_compiler.so` and so most of the tests would require having it installed. The main use-case scenario is when we actually WANT to run tests related to graph compiler so it should be done as easy as possible (without explicitly setting `ENABLE_ELYZOR_GRAPH_COMPILER_LIB_TESTS=1`). 

IMO it's better to see an explicit error instead of a skipped test in case of a missing `libgraph_compiler.so`, as in this case, we're not hiding the problem by default, but rather highlighting it. I believe there should be an explicit action from a user (set env variable) to disable the problematic tests as this is an exceptional case.

Reply:
Sorry, I wasn't aware of `ONEDNN_EXPERIMENTAL_ELYZOR_BACKEND` and this is in fact what I proposed. If it's not defined, the code will not be compiled. It could be defined either by cmake, if the library is found, or by the user. Thus, it's not required to introduce a compile time option, it's already exists.

Reply:
As for me, the hook with the env var, introduced in the last commit, is not required, but it's up to you.
LGTM!

Reply:
removed the hook in ce89359ecacbe2b0d204bb92b5320659d3231072